example/market_order_example.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:io' show Platform;
+
+import 'package:coinbase_cloud_advanced_trade_client/advanced_trade.dart';
+
+final Map<String, String> envVars = Platform.environment;
+final String? apiKeyName = envVars['COINBASE_API_KEY_NAME'];
+final String? privateKeyPEM = envVars['COINBASE_PRIVATE_KEY'];
+
+// This is a simple example of how to use the market order API.
+//
+// To run this example, you will need to create a `Credential` object with your
+// API key and secret. You can get these from the Coinbase Advanced Trade API
+// website.
+//
+// https://docs.cdp.coinbase.com/advanced-trade/docs/welcome/
+//
+// Make sure to set the COINBASE_API_KEY_NAME and COINBASE_PRIVATE_KEY
+// environment variables before running this example.
+//
+// This example will create a market order for BTC-USD for $10 USD.
+void main() async {
+  // Check that the API key and secret are set.
+  if (apiKeyName == null || privateKeyPEM == null) {
+    print('Please set the COINBASE_API_KEY_NAME and COINBASE_PRIVATE_KEY '
+        'environment variables.');
+    return;
+  }
```

Comment:
For better error handling in a command-line application, it's good practice to write error messages to standard error (stderr) and exit with a non-zero status code to signal that an error occurred. This makes it easier to use the script in automated workflows. To do this, you'll also need to update the import at the top of the file from import 'dart:io' show Platform; to import 'dart:io' show Platform, exit, stderr;.

```
  if (apiKeyName == null || privateKeyPEM == null) {
    stderr.writeln('Please set the COINBASE_API_KEY_NAME and COINBASE_PRIVATE_KEY '\n        'environment variables.');
    exit(1);
  }
```